template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/cruisecontrol/CruiseControlRoleConfigProvider.java

```diff
@@ -92,13 +98,21 @@ public boolean sharedRoleType(String roleType) {
                             "com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskCapacityGoal," +
                             "com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkInboundCapacityGoal," +
                             "com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkOutboundCapacityGoal," +
-                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.CpuCapacityGoal"),
+                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.CpuCapacityGoal," +
+                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaDistributionGoal," +
+                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskUsageDistributionGoal," +
+                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.CpuUsageDistributionGoal," +
+                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.TopicReplicaDistributionGoal," +
+                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.LeaderReplicaDistributionGoal"),
                     config(HARD_GOALS, "com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal," +
                             "com.linkedin.kafka.cruisecontrol.analyzer.goals.ReplicaCapacityGoal," +
                             "com.linkedin.kafka.cruisecontrol.analyzer.goals.DiskCapacityGoal," +
                             "com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkInboundCapacityGoal," +
                             "com.linkedin.kafka.cruisecontrol.analyzer.goals.NetworkOutboundCapacityGoal," +
-                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.CpuCapacityGoal")
+                            "com.linkedin.kafka.cruisecontrol.analyzer.goals.CpuCapacityGoal"),
+                    config(AUTHENTICATION_METHOD, "Trusted Proxy"),
+                    config(ADMIN_LEVEL_USERS, "kafka"),
+                    config(TRUSTED_PROXY_SPNEGO_FALLBACK_ENABLED, "true")
             );
```

Comment:
"knox" must be added as trusted proxy user.

Reply:
As discussed, `trusted_auth_service_user=knox` is the default value in the CSD, no need to include it in the provider.